### PR TITLE
[WIP, NO NEED TO MERGE] support migration on BSC network

### PR DESCRIPTION
**Pull request by unclezoro**

### CODE REVIEW

File: contracts/ChildLinkToken.sol [hunk 1]
```diff
@@ -80,4 +85,16 @@ contract ChildLinkToken is
...
   }
+
+  /**
+ * @dev Returns the token owner.
+ */
+  function getOwner() external view returns (address) {
+    return owner();
+  }
+
+  function swap(uint prevTokenAmount) external {
```
Comment (HaoyangLiu) [inline contracts/ChildLinkToken.sol:96]: How about renaming `swap` to `migrate`?

File: flat/ChildLinkToken.sol [hunk 2]
```diff
@@ -296,14 +402,14 @@ library Address {
...
+        // This method relies in extcodesize, which returns 0 for contracts in
+        // construction, since the code is only stored at the end of the
+        // constructor execution.
+
+        uint256 size;
         // solhint-disable-next-line no-inline-assembly
-        assembly { codehash := extcodehash(account) }
-        return (codehash != accountHash && codehash != 0x0);
+        assembly { size := extcodesize(account) }
+        return size > 0;
```
Comment (HaoyangLiu) [inline flat/ChildLinkToken.sol:412]: Why do we need to update this method?

File: contracts/ChildLinkToken.sol (hunk 1 above)
Comment (unclezoro) [reply contracts/ChildLinkToken.sol:96]: `migrate`  maybe not good, the Migration.sol will use it.

Comment (HaoyangLiu) [reply contracts/ChildLinkToken.sol:96]: `Migration.sol` doesn't include `migrate` method. Besides, is there any relationship between `migrate`  and `Migration.sol`?

File: flat/ChildLinkToken.sol (hunk 2 above)
Comment (krebernisak) [reply flat/ChildLinkToken.sol:412]: These are updates from LinkToken v0.6. You can ignore this flat file, but @guagualvcha you should rebase this PR on the updated master branch so we only have the minimal changes here.

**Comment by krebernisak** [comment]
Hey @guagualvcha @HaoyangLiu, I've created a [BSC custom implementation](https://github.com/smartcontractkit/link-token-sidechain-evm/pull/3/files) based on this proposal and the notes/docs you have put forward.

Please take a look and I would appreciate your feedback.

**Pull request closed without merging**